context_prop/src/main/java/io/opentelemetry/context/propagation/HttpTextFormat.java

```diff
@@ -69,15 +69,17 @@
   List<String> fields();
 
   /**
-   * Injects the value downstream. For example, as http headers.
+   * Injects the value downstream. For example, as http headers. The carrier may be null to
+   * facilitate calling this method with a java lambda for the {@link Setter}, in which case that
+   * null will be passed to your {@link Setter} implementation.
    *
    * @param context the {@code Context} containing the value to be injected.
    * @param carrier holds propagation fields. For example, an outgoing message or http request.
    * @param setter invoked for each propagation key to add or remove.
    * @param <C> carrier of propagation fields, such as an http request
    * @since 0.1.0
    */
-  <C> void inject(Context context, C carrier, Setter<C> setter);
+  <C> void inject(Context context, @Nullable C carrier, Setter<C> setter);
```

Comment:
While at it, you could also add `@Nonnull`.

Reply:
is that implied? We haven't done a lot of annotations like that on our parameters. If we want to start annotating all parameters on API methods, let's do that as a separate PR.

Reply:
Right, let's not expand the scope of this PR too much 👌